packages/dashboard-frontend/src/services/helpers/__tests__/sanitizeName.spec.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import sanitizeName from '../sanitizeName';
+
+describe('sanitizeName', () => {
+  test('correct name', () => {
+    const name = 'test-project';
+    const sanitizedName = sanitizeName(name);
+    expect(sanitizedName).toEqual('test-project');
+  });
+
+  test('with uppercase character', () => {
+    const name = 'testProject';
+    const sanitizedName = sanitizeName(name);
+    expect(sanitizedName).toEqual('testproject');
+  });
+
+  test('with non-alphanumeric character in the middle', () => {
+    const name = 'test%-#project';
+    const sanitizedName = sanitizeName(name);
+    expect(sanitizedName).toEqual('test---project');
+  });
+
+  test('with non-alphanumeric first character', () => {
+    const name = '-test-project';
+    const sanitizedName = sanitizeName(name);
+    expect(sanitizedName).toEqual('test-project');
+  });
+
+  test('with non-alphanumeric last character', () => {
+    const name = 'test-project-';
```

Comment:
Should be checked with more than one non-alphanumeric character too. For example: 'test-project--'.